src/backend/aspen/api/authz.py

```diff
@@ -214,3 +216,37 @@ def require_group_privilege(
     privilege: str,
 ) -> AuthorizedGroup:
     return AuthorizedGroup(privilege)
+
+
+class AuthorizedRow:
+    def __init__(self, privilege: str, model: idbase):
+        self.privilege = privilege
+        self.model = model
+
+    async def __call__(
+        self,
+        row_id: int,
+        auth_context: AuthContext = Depends(get_auth_context),
+        session: AsyncSession = Depends(get_db),
+    ) -> idbase:
+        authz_session = AuthZSession(session, auth_context)
+        query = await authz_session.authorized_query(self.privilege, self.model)
+        try:
+            res = (
+                (await (session.execute(query.where(self.model.id == row_id))))
+                .scalars()
+                .one()
+            )
+            return res
+        except NoResultFound:
+            raise ex.UnauthorizedException(f"unauthorized")
+
+
+# NOTE - any endpoint that uses `fetch_authorized_row` must use "row_id" as the path
+#        parameter name to identify the row to be fetched.
+@lru_cache
+def fetch_authorized_row(
```

Comment:
A shortcut for endpoints that try to read/write/etc a particular item to just get the item they are about, or return an Unauthorized exception if it doesn't exist or they don't have the required permissions.

Reply:
What is a `row_id`? Is it literally the `id` column in the `groups` table?

Reply:
if the model object passed in is a group, yes. It refers to the `id` column on whatever `model` class gets passed in. Thoughts on whether I can make this clearer?